File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/places/raids/Kaunis.java
package it.paranoidsquirrels.idleguildmaster.storage.data.places.raids;

import it.paranoidsquirrels.idleguildmaster.AchievementsUtils;
import it.paranoidsquirrels.idleguildmaster.MainActivity;
import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.databinding.LayoutDungeonBinding;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy;
import it.paranoidsquirrels.idleguildmaster.storage.data.places.Area;
import it.paranoidsquirrels.idleguildmaster.storage.data.places.Logger;
import java.util.Arrays;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.concurrent.CopyOnWriteArrayList;

/* JADX INFO: loaded from: classes3.dex */
public class Kaunis extends Area {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public int adventurersNumber() {
        return 14;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public int costToRefresh() {
        return 15;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public int getAreaType() {
        return 1;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public int getDarkness() {
        return 18;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public int getName() {
        return R.string.raid_name_kaunis;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public int getSummaryDrawable() {
        return R.drawable.summary_kaunis;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public int getDetailDrawable() {
        return R.drawable.area_kaunis;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public LayoutDungeonBinding getLayout() {
        return MainActivity.raidsFragment.getBinding().kaunis;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    protected List<Enemy> rollEnemies() {
        int i = this.progress;
        if (i == 1) {
            return new CopyOnWriteArrayList(Arrays.asList(Enemy.getInstance("Necrobot"), Enemy.getInstance("Necrobot"), Enemy.getInstance("Necrobot")));
        }
        if (i == 6) {
            return new CopyOnWriteArrayList(Arrays.asList(Enemy.getInstance("Necrobot"), Enemy.getInstance("Necrobot"), Enemy.getInstance("Enforcer"), Enemy.getInstance("Necrobot"), Enemy.getInstance("Necrobot")));
        }
        if (i != 16) {
            switch (i) {
                case 9:
                    return new CopyOnWriteArrayList(Arrays.asList(Enemy.getInstance("Phantasm")));
                case 10:
                    return new CopyOnWriteArrayList(Arrays.asList(Enemy.getInstance("Necrobot"), Enemy.getInstance("Enforcer"), Enemy.getInstance("Enforcer"), Enemy.getInstance("Necrobot")));
                case 11:
                    return new CopyOnWriteArrayList(Arrays.asList(Enemy.getInstance("Necrobot"), Enemy.getInstance("Necrobot"), Enemy.getInstance("Cerebrum"), Enemy.getInstance("Necrobot"), Enemy.getInstance("Necrobot")));
                case 12:
                    return new CopyOnWriteArrayList(Arrays.asList(Enemy.getInstance("Necrobot"), Enemy.getInstance("Phantasm"), Enemy.getInstance("Necrobot")));
                default:
                    return new CopyOnWriteArrayList();
            }
        }
        return new CopyOnWriteArrayList(Arrays.asList(Enemy.getInstance("ChiefScientistAva"), Enemy.getInstance("KingAino"), Enemy.getInstance("FirstMinisterAtos")));
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    protected void searchRoom() {
        Logger.log(this, 41, new Object[0]);
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    protected void triggerEvent(String str) {
        str.hashCode();
        switch (str) {
            case "kill_KingAino":
            case "kill_ChiefScientistAva":
            case "kill_FirstMinisterAtos":
                AchievementsUtils.unlock(AchievementsUtils.ACHIEVEMENT_THE_COUNCIL);
                break;
            case "enter_room":
                switch (this.progress) {
                    case 1:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_room_1));
                        break;
                    case 2:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_room_2));
                        break;
                    case 3:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_room_3));
                        break;
                    case 4:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_room_4));
                        break;
                    case 5:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_room_5));
                        break;
                    case 6:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_room_6));
                        break;
                    case 7:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_room_7));
                        break;
                    case 8:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_room_8));
                        break;
                    case 9:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_room_9));
                        break;
                    case 10:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_room_10));
                        break;
                    case 11:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_room_11));
                        break;
                    case 12:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_room_12));
                        break;
                    case 13:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_room_13));
                        break;
                    case 14:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_room_14));
                        break;
                    case 15:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_room_15));
                        break;
                    case 16:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_room_16));
                        break;
                    case 17:
                        Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_room_17));
                        this.terminationRequested = true;
                        break;
                }
                            case "fight_start":
                int i = this.progress;
                if (i == 1) {
                    Logger.log(this, 101, Integer.valueOf(R.string.log_kaunis_encounter_1));
                    break;
                } else if (i == 6) {
                    Logger.log(this, 101, Integer.valueOf(R.string.log_kaunis_encounter_2));
                    break;
                } else if (i != 16) {
                    switch (i) {
                        case 9:
                            Logger.log(this, 101, Integer.valueOf(R.string.log_kaunis_encounter_3));
                            break;
                        case 10:
                            Logger.log(this, 101, Integer.valueOf(R.string.log_kaunis_encounter_4));
                            break;
                        case 11:
                            Logger.log(this, 101, Integer.valueOf(R.string.log_kaunis_encounter_5));
                            break;
                        case 12:
                            Logger.log(this, 101, Integer.valueOf(R.string.log_kaunis_encounter_6));
                            break;
                    }
                } else {
                    Logger.log(this, 101, Integer.valueOf(R.string.log_kaunis_encounter_7));
                    break;
                }
                            case "enter_dungeon":
                Logger.log(this, 100, Integer.valueOf(R.string.log_kaunis_enter));
                break;
        }
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public LinkedHashMap<Area, Integer> listAreasUnlocked() {
        return new LinkedHashMap<>();
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
    public List<Enemy> listEnemies() {
        return Arrays.asList(Enemy.getInstance("Necrobot"), Enemy.getInstance("Enforcer"), Enemy.getInstance("Phantasm"), Enemy.getInstance("Cerebrum"), Enemy.getInstance("ChiefScientistAva"), Enemy.getInstance("KingAino"), Enemy.getInstance("FirstMinisterAtos"));
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/places/raids/Kaunis.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.places.raids

import it.paranoidsquirrels.idleguildmaster.AchievementsUtils
import it.paranoidsquirrels.idleguildmaster.MainActivity
import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.databinding.LayoutDungeonBinding
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
import it.paranoidsquirrels.idleguildmaster.storage.data.places.Area
import it.paranoidsquirrels.idleguildmaster.storage.data.places.Logger
import java.util.LinkedHashMap
import java.util.concurrent.CopyOnWriteArrayList

class Kaunis : Area() {
    override fun adventurersNumber(): Int = 14

    override fun costToRefresh(): Int = 15

    override fun getAreaType(): Int = 1

    override fun getDarkness(): Int = 18

    override fun getName(): Int = R.string.raid_name_kaunis

    override fun getSummaryDrawable(): Int = R.drawable.summary_kaunis

    override fun getDetailDrawable(): Int = R.drawable.area_kaunis

    override fun getLayout(): LayoutDungeonBinding = MainActivity.raidsFragment.binding!!.kaunis

    override fun rollEnemies(): MutableList<Enemy> {
        return when (progress) {
            1 -> CopyOnWriteArrayList(listOfNotNull(
                Enemy.getInstance("Necrobot"),
                Enemy.getInstance("Necrobot"),
                Enemy.getInstance("Necrobot")
            ))
            6 -> CopyOnWriteArrayList(listOfNotNull(
                Enemy.getInstance("Necrobot"),
                Enemy.getInstance("Necrobot"),
                Enemy.getInstance("Enforcer"),
                Enemy.getInstance("Necrobot"),
                Enemy.getInstance("Necrobot")
            ))
            9 -> CopyOnWriteArrayList(listOfNotNull(Enemy.getInstance("Phantasm")))
            10 -> CopyOnWriteArrayList(listOfNotNull(
                Enemy.getInstance("Necrobot"),
                Enemy.getInstance("Enforcer"),
                Enemy.getInstance("Enforcer"),
                Enemy.getInstance("Necrobot")
            ))
            11 -> CopyOnWriteArrayList(listOfNotNull(
                Enemy.getInstance("Necrobot"),
                Enemy.getInstance("Necrobot"),
                Enemy.getInstance("Cerebrum"),
                Enemy.getInstance("Necrobot"),
                Enemy.getInstance("Necrobot")
            ))
            12 -> CopyOnWriteArrayList(listOfNotNull(
                Enemy.getInstance("Necrobot"),
                Enemy.getInstance("Phantasm"),
                Enemy.getInstance("Necrobot")
            ))
            16 -> CopyOnWriteArrayList(listOfNotNull(
                Enemy.getInstance("ChiefScientistAva"),
                Enemy.getInstance("KingAino"),
                Enemy.getInstance("FirstMinisterAtos")
            ))
            else -> CopyOnWriteArrayList()
        }
    }

    override fun searchRoom() {
        Logger.log(this, 41)
    }

    override fun triggerEvent(str: String) {
        when (str) {
            "kill_KingAino", "kill_ChiefScientistAva", "kill_FirstMinisterAtos" -> {
                AchievementsUtils.unlock(AchievementsUtils.ACHIEVEMENT_THE_COUNCIL)
            }
            "enter_room" -> {
                when (progress) {
                    1 -> Logger.log(this, 100, R.string.log_kaunis_room_1)
                    2 -> Logger.log(this, 100, R.string.log_kaunis_room_2)
                    3 -> Logger.log(this, 100, R.string.log_kaunis_room_3)
                    4 -> Logger.log(this, 100, R.string.log_kaunis_room_4)
                    5 -> Logger.log(this, 100, R.string.log_kaunis_room_5)
                    6 -> Logger.log(this, 100, R.string.log_kaunis_room_6)
                    7 -> Logger.log(this, 100, R.string.log_kaunis_room_7)
                    8 -> Logger.log(this, 100, R.string.log_kaunis_room_8)
                    9 -> Logger.log(this, 100, R.string.log_kaunis_room_9)
                    10 -> Logger.log(this, 100, R.string.log_kaunis_room_10)
                    11 -> Logger.log(this, 100, R.string.log_kaunis_room_11)
                    12 -> Logger.log(this, 100, R.string.log_kaunis_room_12)
                    13 -> Logger.log(this, 100, R.string.log_kaunis_room_13)
                    14 -> Logger.log(this, 100, R.string.log_kaunis_room_14)
                    15 -> Logger.log(this, 100, R.string.log_kaunis_room_15)
                    16 -> Logger.log(this, 100, R.string.log_kaunis_room_16)
                    17 -> {
                        Logger.log(this, 100, R.string.log_kaunis_room_17)
                        terminationRequested = true
                    }
                }
            }
            "fight_start" -> {
                when (progress) {
                    1 -> Logger.log(this, 101, R.string.log_kaunis_encounter_1)
                    6 -> Logger.log(this, 101, R.string.log_kaunis_encounter_2)
                    9 -> Logger.log(this, 101, R.string.log_kaunis_encounter_3)
                    10 -> Logger.log(this, 101, R.string.log_kaunis_encounter_4)
                    11 -> Logger.log(this, 101, R.string.log_kaunis_encounter_5)
                    12 -> Logger.log(this, 101, R.string.log_kaunis_encounter_6)
                    16 -> Logger.log(this, 101, R.string.log_kaunis_encounter_7)
                }
            }
            "enter_dungeon" -> {
                Logger.log(this, 100, R.string.log_kaunis_enter)
            }
        }
    }

    override fun listAreasUnlocked(): LinkedHashMap<Area, Int> = LinkedHashMap()

    override fun listEnemies(): List<Enemy> {
        return listOfNotNull(
            Enemy.getInstance("Necrobot"),
            Enemy.getInstance("Enforcer"),
            Enemy.getInstance("Phantasm"),
            Enemy.getInstance("Cerebrum"),
            Enemy.getInstance("ChiefScientistAva"),
            Enemy.getInstance("KingAino"),
            Enemy.getInstance("FirstMinisterAtos")
        )
    }
}
